File: ExamenUnidad2_PYTHON 1/ExamenUnidad2_PYTHON 1/ExamenUnidad2/Animales.py
class Animales:
    IDENTIFICADOR2 = 1000

    def __init__(self, tipo_animal, fecha_de_nacimiento, fecha_de_llegada, enfermedad, tipo_de_alimentacion,
                 peso, frecuencia_de_alimentacion, cuenta_con_vacunas):
        self.tipo_animal = tipo_animal
        self.fecha_de_llegada = fecha_de_llegada
        self.enfermedad = enfermedad
        self.tipo_de_alimentacion = tipo_de_alimentacion
        self.fecha_de_nacimiento = fecha_de_nacimiento
        self.peso = peso
        self.frecuencia_de_alimentacion = frecuencia_de_alimentacion
        self.cuenta_con_vacunas = cuenta_con_vacunas
        self.id = Animales.IDENTIFICADOR2
        Animales.IDENTIFICADOR2 += 1
    
    def set_identificador2(identificador2):
        Animales.IDENTIFICADOR_2 = identificador2

    def get_id(self):
        return self.id

    def set_id(self, id):
        self.id = id

    def get_tipo_animal(self):
        return self.tipo_animal

    def set_tipo_animal(self, tipo_animal):
        self.tipo_animal = tipo_animal

    def get_fecha_de_llegada(self):
        return self.fecha_de_llegada

    def set_fecha_de_llegada(self, fecha_de_llegada):
        self.fecha_de_llegada = fecha_de_llegada

    def get_enfermedad(self):
        return self.enfermedad

    def set_enfermedad(self, enfermedad):
        self.enfermedad = enfermedad

    def get_tipo_de_alimentacion(self):
        return self.tipo_de_alimentacion

    def set_tipo_de_alimentacion(self, tipo_de_alimentacion):
        self.tipo_de_alimentacion = tipo_de_alimentacion

    def get_fecha_de_nacimiento(self):
        return self.fecha_de_nacimiento

    def set_fecha_de_nacimiento(self, fecha_de_nacimiento):
        self.fecha_de_nacimiento = fecha_de_nacimiento

    def get_peso(self):
        return self.peso

    def set_peso(self, peso):
        self.peso = peso

    def get_frecuencia_de_alimentacion(self):
        return self.frecuencia_de_alimentacion

    def set_frecuencia_de_alimentacion(self, frecuencia_de_alimentacion):
        self.frecuencia_de_alimentacion = frecuencia_de_alimentacion

    def is_cuenta_con_vacunas(self):
        return self.cuenta_con_vacunas

    def set_cuenta_con_vacunas(self, cuenta_con_vacunas):
        self.cuenta_con_vacunas = cuenta_con_vacunas 


    

    def obtener_informacion_animal(self):
        enfermedades_string = ", ".join(self.enfermedad) if self.enfermedad else "Ninguna"
        return (f"\nId: {self.id} \nTipo de animal: {self.tipo_animal} \nFecha de nacimiento: {self.fecha_de_nacimiento} \n"
        f"Fecha de llegada al zoológico: {self.fecha_de_llegada} \nPeso: {self.peso:.2f} \nEnfermedades: {enfermedades_string} \n"
        f"Frecuencia de alimentación: {self.frecuencia_de_alimentacion} \nTipo de alimentación: {self.tipo_de_alimentacion}")

File: ExamenUnidad2__JAVA/ExamenUnidad2/src/examenunidad2/Animales.java
package examenunidad2;


public class Animales {

    private static int IDENTIFICADOR2 = 1000;
    private int id;
    private String tipoAnimal;
    private String fechaDeLlegada;
    private String[] enfermedad;
    private String tipoDeAlimentacion;
    private String fechaDeNacimiento;
    private double peso;
    private int frecuenciaDeAlimentacion;
    private boolean cuentaConVacunas;

    public Animales(String tipoAnimal, String fechaDeNacimiento, String fechaDeLlegada, String[] enfermedad,
            String tipoDeAlimentacion, double peso,
            int frecuenciaDeAlimentacion, boolean cuentaConVacunas) {
        this.tipoAnimal = tipoAnimal;
        this.fechaDeLlegada = fechaDeLlegada;
        this.enfermedad = enfermedad;
        this.tipoDeAlimentacion = tipoDeAlimentacion;
        this.fechaDeNacimiento = fechaDeNacimiento;
        this.peso = peso;
        this.frecuenciaDeAlimentacion = frecuenciaDeAlimentacion;
        this.cuentaConVacunas = cuentaConVacunas;
        this.id = IDENTIFICADOR2 +=1;
    }

    public static int getIDENTIFICADOR2() {
        return IDENTIFICADOR2;
    }

    public static void setIDENTIFICADOR2(int IDENTIFICADOR2) {
        Animales.IDENTIFICADOR2 = IDENTIFICADOR2;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTipoAnimal() {
        return tipoAnimal;
    }

    public void setTipoAnimal(String tipoAnimal) {
        this.tipoAnimal = tipoAnimal;
    }

    public String getFechaDeLlegada() {
        return fechaDeLlegada;
    }

    public void setFechaDeLlegada(String fechaDeLlegada) {
        this.fechaDeLlegada = fechaDeLlegada;
    }

    public String[] getEnfermedad() {
        return enfermedad;
    }

    public void setEnfermedad(String[] enfermedad) {
        this.enfermedad = enfermedad;
    }

    

    public String getTipoDeAlimentacion() {
        return tipoDeAlimentacion;
    }

    public void setTipoDeAlimentacion(String tipoDeAlimentacion) {
        this.tipoDeAlimentacion = tipoDeAlimentacion;
    }

    public String getFechaDeNacimiento() {
        return fechaDeNacimiento;
    }

    public void setFechaDeNacimiento(String fechaDeNacimiento) {
        this.fechaDeNacimiento = fechaDeNacimiento;
    }

    public double getPeso() {
        return peso;
    }

    public void setPeso(double peso) {
        this.peso = peso;
    }

    public int getFrecuenciaDeAlimentacion() {
        return frecuenciaDeAlimentacion;
    }

    public void setFrecuenciaDeAlimentacion(int frecuenciaDeAlimentacion) {
        this.frecuenciaDeAlimentacion = frecuenciaDeAlimentacion;
    }

    public boolean isCuentaConVacunas() {
        return cuentaConVacunas;
    }

    public void setCuentaConVacunas(boolean cuentaConVacunas) {
        this.cuentaConVacunas = cuentaConVacunas;
    }
    
    public String obtenerInformacionAnimal()
    {
        StringBuilder enfermedadesString = new StringBuilder();
    if (enfermedad != null) {
        for (String e : enfermedad) {
            enfermedadesString.append(e).append(", ");
        }
    }
       return String.format("\nId: %d \nTipo de animal: %s \nFecha de nacimiento: %s \nFecha de llegada al zoológico: %s \nPeso: %.2f \nEnfermedades: %s \nFrecuencia de alimentación: %d \nTipo de alimentación: %s \nCuenta con vacunas: %b",
            id, tipoAnimal, fechaDeNacimiento, fechaDeLlegada, peso, enfermedadesString, frecuenciaDeAlimentacion, tipoDeAlimentacion, cuentaConVacunas);
}
    
    
    
}
